src/socket.c

```diff
@@ -0,0 +1,91 @@
+#include <zephyr/logging/log.h>
+
+#include "tb_internal.h"
+
+LOG_MODULE_DECLARE(thingsboard_client, CONFIG_THINGSBOARD_LOG_LEVEL);
+
+static struct sockaddr_storage thingsboard_server_address;
+
+int thingsboard_server_resolve(const char *hostname, uint16_t port, struct sockaddr_storage *server)
+{
+	int err;
+	struct zsock_addrinfo *result = NULL;
```

Comment:
according to https://pubs.opengroup.org/onlinepubs/9699919799/functions/getaddrinfo.html this is actually a result set, maybe 
```suggestion
	struct zsock_addrinfo *results = NULL;
```
?